packages/react-reconciler/src/ReactFiberHydrationContext.old.js

```diff
@@ -538,10 +538,9 @@ function popHydrationState(fiber: Fiber): boolean {
       (shouldDeleteUnhydratedTailInstances(fiber.type) &&
         !shouldSetTextContent(fiber.type, fiber.memoizedProps)))
   ) {
-    let nextInstance = nextHydratableInstance;
-    while (nextInstance) {
-      deleteHydratableInstance(fiber, nextInstance);
-      nextInstance = getNextHydratableSibling(nextInstance);
+    const nextInstance = nextHydratableInstance;
+    if (nextInstance) {
+      throwOnHydrationMismatchIfConcurrentMode(fiber);
```

Comment:
This only throws if it's "concurrent mode" but if it's not, it should still fall through to the old logic, no?